### PR TITLE
Hide global configuration and system information from non super users

### PR DESCRIPTION
**Pull request by Quy**
Pull Request for Issue #19478.

### Summary of Changes
Only super users have access to `Global Configuration` and `System Information`. This PR hides these 2 menu items from administrators.

Use identical operators for minor performance improvement.

### Testing Instructions
Log in as an Administrator.
View `System` menu.

### Expected result
No `Global Configuration` and `System Information`

### Actual result
Displays `Global Configuration` and `System Information`

### Documentation Changes Required
none

### CODE REVIEW

File: administrator/modules/mod_menu/menu.php [hunk 1]
```diff
@@ -314,6 +314,19 @@ protected function preprocess($items)
 
 				list($assetName) = isset($query['context']) ? explode('.', $query['context'], 2) : array('com_fields');
 			}
+			elseif ($item->element == 'com_config' && !$user->authorise('core.admin'))
```
Comment (C-Lodder) [inline administrator/modules/mod_menu/menu.php]: `===`

File: administrator/modules/mod_menu/menu.php [hunk 2]
```diff
@@ -314,6 +314,19 @@ protected function preprocess($items)
...
+			{
+				continue;
+			}
+			elseif ($item->element == 'com_admin')
```
Comment (C-Lodder) [inline administrator/modules/mod_menu/menu.php]: `===`

File: administrator/modules/mod_menu/menu.php [hunk 3]
```diff
@@ -314,6 +314,19 @@ protected function preprocess($items)
...
+			{
+				parse_str($item->link, $query);
+
+				if (isset($query['view']) && $query['view'] == 'sysinfo' && !$user->authorise('core.admin'))
+				{
+					continue;
+				}
+			}
 
 			if ($assetName && !$user->authorise(($item->scope == 'edit') ? 'core.create' : 'core.manage', $assetName))
```
Comment (C-Lodder) [inline administrator/modules/mod_menu/menu.php]: `===`

File: administrator/modules/mod_menu/menu.php [hunk 4]
```diff
@@ -314,6 +314,19 @@ protected function preprocess($items)
...
 			}
+			elseif ($item->element == 'com_config' && !$user->authorise('core.admin'))
+			{
+				continue;
+			}
+			elseif ($item->element == 'com_admin')
+			{
+				parse_str($item->link, $query);
+
+				if (isset($query['view']) && $query['view'] == 'sysinfo' && !$user->authorise('core.admin'))
```
Comment (C-Lodder) [inline administrator/modules/mod_menu/menu.php]: `===`

**Comment by dautrich** [comment]
I have tested this item :white_check_mark: successfully on 506a66c9bc9a261cc7c346b84f497215c42d8106<br /><br />Logged in as **Administrator**: Menu items **Global Configuration** and **System Information** _not_ shown

Logged in as **Super User**: Menu items **Global Configuration** and **System Information** shown<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19697">issues.joomla.org/tracker/joomla-cms/19697</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on dde09670bc9016b245572e4bd35e5ecfeb0a3ded<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19697">issues.joomla.org/tracker/joomla-cms/19697</a>.</sub>

**Comment by alikon** [comment]
however i've noticed a small inconsistence (with or without this pr)
as an administrator
on: **system information** http://localhost/test/administrator/index.php?option=com_admin&view=sysinfo you got
screenshot from 2018-02-17 09-51-29
 
 `throw new JAccessExceptionNotallowed(JText::_('JERROR_ALERTNOAUTHOR'), 403);`

on: **configuration** http://localhost/test/administrator/index.php?option=com_config

screenshot from 2018-02-17 09-55-28

`$this->app->enqueueMessage(JText::_('JERROR_ALERTNOAUTHOR'));`

**Comment by ghost** [comment]
@alikon can you please open a new Issue for the small inconsistence you reported above?

**Comment by ghost** [comment]
@dautrich can you please retest?

**Comment by dautrich** [comment]
I have tested this item :white_check_mark: successfully on dde09670bc9016b245572e4bd35e5ecfeb0a3ded<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19697">issues.joomla.org/tracker/joomla-cms/19697</a>.</sub>

**Comment by dautrich** [comment]
Done<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19697">issues.joomla.org/tracker/joomla-cms/19697</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19697">issues.joomla.org/tracker/joomla-cms/19697</a>.</sub>

**Pull request merged**